src/unitxt/metrics.py

```diff
@@ -367,16 +437,245 @@ def compute(
 
 
 class InstanceMetric(SingleStreamOperator, MetricWithConfidenceInterval):
+    """Class for metrics for which a global score can be calculated by aggregating the instance scores (possibly with additional instance inputs).
+
+    InstanceMetric currently allows three reductions:
+    1. 'mean', which calculates the mean of instance scores,'
+    2. 'group_mean', which first applies an aggregation function specified in the reduction_map
+        to instance scores grouped by the field grouping_field (which must not be None), and returns the mean
+        of the group scores; if grouping_field is None, grouping is disabled.
+        See _validate_group_mean_reduction for formatting instructions.
+    3. 'group_mean_subgroup_comparison': compare sub-groups (e.g. a baseline and others) within
+        groups, then return the mean of this function value.
+        See _validate_group_mean_subgroup_comparison_reduction for formatting instructions.
+    """
+
     n_resamples = _N_RESAMPLES_DEFAULT_FOR_INSTANCE_METRICS
 
-    implemented_reductions: List[str] = field(default_factory=lambda: ["mean"])
+    implemented_reductions: List[str] = field(
+        default_factory=lambda: ["mean", "group_mean", "group_mean_subgroup_comparison"]
+    )
+    grouping_field: str = None
 
     @property
     @abstractmethod
     def reduction_map(self) -> dict:
         pass
 
+    def _validate_group_mean_reduction(self, reduction_name="group_mean"):
+        """Ensure that group_mean reduction_map is properly formatted.
+
+        Example: Apply the variance (np.var) to group Accuracy instance scores.  This class would be specified as follows:
+
+        class GroupVarianceAccuracy(Accuracy):
+            grouping_field = 'group_id'
+            reduction_map = {'group_mean': {'agg_func': ['variance', np.var, True]}}
+
+        reduction_map must be a dict with
+        - an 'agg_func' field with value being a 3-element list where
+            - 1st element is a string name of the aggregation function (used in naming the CI report)
+            - 2nd element is the callable aggregation function
+            - 3rd element is a Boolean indicator of whether, during boostrap CI calculation, the groups are to be sampled as single units.
+                If True, the group scores are calculated and then resampled.  This treats the group units as the unit of
+                interest for which the CI is being compared.
+                If False, the instances are resampled individually, and the groups determined
```

Comment:
In our specific use case, we have groups that consist of an observation that is the original, and several that are paraphrases of the original sentence.   If the grouping is not enforced (False), then instances are resampled individually and then grouped.  Because the comparison aggregation functions require both baseline and other inputs, this has several effects.  Primarily, if the "original" instance for a given group is not resampled, we can't calculate the function at all.  Therefore to preserve the information, since the unit of interest is "an original sentence and its paraphrase" then we want to resample the units.